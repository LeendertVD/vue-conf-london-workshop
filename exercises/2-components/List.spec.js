import List from './List.vue'
import { shallowMount } from '@vue/test-utils'

test('renders "no items" if items prop is undefined', () => {
    // Arrange
    const wrapper = shallowMount(List)
    // Act

    // Assert
    expect(wrapper.text()).toContain('No items')

})

test('renders text using each item in items prop array', () => {
    // Arrange
    const wrapper = shallowMount(List, {
        propsData: {
            items: ['item1', 'item2']
        }
    })
    // Act

    // Assert
    expect(wrapper.text()).toContain('item1')
    expect(wrapper.text()).toContain('item2')
})

test('renders text using each item in items prop array - snapshot', () => {
    // Arrange
    const wrapper = shallowMount(List, {
        propsData: {
            items: ['item1', 'item2']
        }
    })

    // Act
    expect(wrapper.element).toMatchSnapshot();
})
