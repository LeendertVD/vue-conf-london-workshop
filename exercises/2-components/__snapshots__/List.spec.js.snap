// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text using each item in items prop array - snapshot 1`] = `
<ul>
  <li>
    
    item1
  
  </li>
  <li>
    
    item2
  
  </li>
</ul>
`;
